pandas/core/groupby/groupby.py

```diff
@@ -1157,7 +1159,8 @@ def mean(self, *args, **kwargs):
         """
         nv.validate_groupby_func('mean', args, kwargs, ['numeric_only'])
         try:
-            return self._cython_agg_general('mean', **kwargs)
+            return self._cython_agg_general(
```

Comment:
Is there a particular reason why we don't reuse the `groupby_function` factory (save perhaps current scoping) for these? Wondering why we would do `Series(x).mean(**kwargs)` here instead of `np.mean` as we do with the existing funcs

Reply:
``np.mean`` does not properly handle argument, for example ``min_count``

Reply:
Gotcha. So we probably have an issue with `sum` as generated by that function then?

Don't need to necessarily solve here just trying to get an understanding of how we can align all of these defs

Reply:
likely we should re-write ``grouping_function``, but its behavior is actually kind of tricky as it handles cases where the alt_function raises and must fallback; for another time.